extensions/functions_comparison.yaml

```diff
@@ -73,4 +73,12 @@ scalar_functions:
           - value: any1
         return: BOOLEAN
         nullability: DECLARED_OUTPUT
+  -
+    name: "is_nan"
+    description: Whether a value is not a number.
+    impls:
+      - args:
+          - value: any1
```

Comment:
This doesn't seem like the right argument type. Should we just have two of these: one for fp32 and one for fp64?

Reply:
It appears Arrow supports all numeric types but for anything other than float or double the return is hard-coded to false.  I'd be onboard with only supporting floating point types.